tests/test_deserialization.py

```diff
@@ -410,48 +410,114 @@ def test_invalid_datetime_deserialization(self, in_value):
         with pytest.raises(ValidationError, match="Not a valid datetime."):
             field.deserialize(in_value)
 
-    def test_datetime_passed_year_is_invalid(self):
-        field = fields.DateTime()
-        with pytest.raises(ValidationError):
-            field.deserialize("1916")
-
-    def test_datetime_passed_date_is_invalid(self):
-        field = fields.DateTime()
-        with pytest.raises(ValidationError):
-            field.deserialize("2017-04-13")
-
     def test_custom_date_format_datetime_field_deserialization(self):
-        dtime = dt.datetime.now()
-        datestring = dtime.strftime("%H:%M:%S.%f %Y-%m-%d")
+        # Datetime string with format "%H:%M:%S.%f %Y-%m-%d"
+        datestring = "10:11:12.123456 2019-01-02"
 
+        # Deserialization should fail when datestring is not of same format
         field = fields.DateTime(format="%d-%m-%Y %H:%M:%S")
-        # deserialization should fail when datestring is not of same format
         with pytest.raises(ValidationError, match="Not a valid datetime."):
             field.deserialize(datestring)
+
         field = fields.DateTime(format="%H:%M:%S.%f %Y-%m-%d")
-        assert_datetime_equal(field.deserialize(datestring), dtime)
+        assert field.deserialize(datestring) == dt.datetime(2019, 1, 2, 10, 11, 12, 123456)
+
+        field = fields.NaiveDateTime(format="%H:%M:%S.%f %Y-%m-%d")
+        assert field.deserialize(datestring) == dt.datetime(2019, 1, 2, 10, 11, 12, 123456)
+
+        field = fields.AwareDateTime(format="%H:%M:%S.%f %Y-%m-%d")
+        with pytest.raises(ValidationError) as excinfo:
+            field.deserialize(datestring)
+        assert excinfo.value.args[0] == "Not a valid aware datetime."
```

Comment:
Nit: Passing `match` to `pytest.raises` is tad nicer and more consistent with the other tests.